--- a/Triangulation_3/include/CGAL/Delaunay_triangulation_3.h	2024-10-19 00:04:13.000000000 +0800
+++ b/Triangulation_3/include/CGAL/Delaunay_triangulation_3.h	2025-04-12 21:27:24.623557700 +0800
@@ -411,6 +411,19 @@
   std::ptrdiff_t insert_with_info(InputIterator first, InputIterator last)
   {
     size_type n = number_of_vertices();
     std::vector<std::size_t> indices;
     std::vector<Point> points;
     std::vector<typename Triangulation_data_structure::Vertex::Info> infos;
+    using iterator_category = typename std::iterator_traits<InputIterator>::iterator_category;
+    constexpr bool can_reserve = std::is_base_of<std::forward_iterator_tag, iterator_category>::value;
+    if (can_reserve) {
+      const size_t point_count = static_cast<size_t>(std::distance(first, last));
+      indices.reserve(point_count);
+      points.reserve(point_count);
+      infos.reserve(point_count);
+    }
@@ -422,6 +435,13 @@
       infos.push_back(top_get_second(value));
       indices.push_back(index++);
     }
 
     typedef typename Pointer_property_map<Point>::type Pmap;
     typedef Spatial_sort_traits_adapter_3<Geom_traits,Pmap> Search_traits;
